async-query/src/test/java/org/opensearch/sql/spark/scheduler/job/OpenSearchRefreshIndexJobTest.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.sql.spark.scheduler.job;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertSame;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.spy;
+import static org.mockito.Mockito.verify;
+
+import java.time.Instant;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Answers;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.opensearch.client.Client;
+import org.opensearch.cluster.service.ClusterService;
+import org.opensearch.jobscheduler.spi.JobExecutionContext;
+import org.opensearch.jobscheduler.spi.ScheduledJobParameter;
+import org.opensearch.sql.spark.scheduler.model.OpenSearchRefreshIndexJobRequest;
+import org.opensearch.threadpool.ThreadPool;
+
+public class OpenSearchRefreshIndexJobTest {
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private ClusterService clusterService;
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private ThreadPool threadPool;
+
+  @Mock(answer = Answers.RETURNS_DEEP_STUBS)
+  private Client client;
+
+  @Mock private JobExecutionContext context;
+
+  private OpenSearchRefreshIndexJob jobRunner;
+
+  private OpenSearchRefreshIndexJob spyJobRunner;
+
+  @BeforeEach
+  public void setup() {
+    MockitoAnnotations.openMocks(this);
+    jobRunner = OpenSearchRefreshIndexJob.getJobRunnerInstance();
+    jobRunner.setClient(null);
+    jobRunner.setClusterService(null);
+    jobRunner.setThreadPool(null);
```

Comment:
why use setXXX? can we construct OpenSearchRefreshIndexJob(client, clusterService, threadpool)?

Reply:
> why use setXXX? can we construct OpenSearchRefreshIndexJob(client, clusterService, threadpool)?

Explained on comments here https://github.com/opensearch-project/sql/pull/2834/files/2b69b66166dcc0b022404d7424717b6503265bef#diff-b63487ae812d0eeb5c84aa6be4bca723773ffbe2f04056ff1807d31ca4d0e67eR23-R31